skyportal/handlers/api/analysis.py

```diff
@@ -54,6 +57,18 @@
 
 DEFAULT_ANALYSES_DAILY_LIMIT = 100
 
+# check for API key
+summary_config = copy.deepcopy(cfg['analysis_services.openai_analysis_service.summary'])
+if summary_config.get("api_key"):
+    # there may be a global API key set in the config file
+    openai_api_key = summary_config.get("api_key")
+    summary_config.pop("api_key")
```

Comment:
Can we combine these two lines into just "pop"?

Reply:
We use the summary_config later on.

Reply:
yeah, do `openai_api_key=summary_config.pop('api_key')`

Reply:
Ah got ya. Done.